Source/DiabloUI/selhero.cpp

```diff
@@ -565,6 +565,13 @@ static void UiSelHeroDialog(
 		while (!selhero_endMenu && !selhero_navigateYesNo) {
 			UiClearScreen();
 			UiRenderItems(vecSelHeroDialog);
+			std::vector<std::unique_ptr<UiItemBase>> tmp;
+			const Point uiPosition = GetUIRectangle().position;
+			for (int i = 0; i < selhero_heroInfo.herorank; i++) {
+				SDL_Rect rect = { (Sint16)(uiPosition.x + 30) + i * 12, (Sint16)(uiPosition.y + 275), 12, 12 };
+				tmp.push_back(std::make_unique<UiImageAnimatedClx>(*DifficultyIndicator, rect, UiFlags::None));

```

Comment:
That creates a vector on every frame. Let's just create / update it on hero change?